spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,351 @@
+
+            wait_for { `pgrep -f libdatadog-crashtracking-receiver`.lines.size }.to be 1
+
+            tear_down!
+          end
+        end
+
+        context 'when forked' do
+          it 'starts a second crash tracker for the fork' do
+            crashtracker = build_crashtracker
+
+            crashtracker.start
+
+            expect_in_fork do
+              wait_for { `pgrep -f libdatadog-crashtracking-receiver`.lines.size }.to be 2
+            end
+
+            tear_down!
+          end
+        end
+      end
+
+      describe '#stop' do
+        context 'when _native_stop_crashtracker raises an exception' do
+          it 'logs the exception' do
+            logger = Logger.new($stdout)
+            crashtracker = build_crashtracker(logger: logger)
+
+            expect(described_class).to receive(:_native_stop) { raise 'Test failure' }
+            expect(logger).to receive(:error).with(/Failed to stop crash tracking: Test failure/)
+
+            crashtracker.stop
+          end
+        end
+
+        it 'stops the crash tracker' do
+          crashtracker = build_crashtracker
+
+          crashtracker.start
+
+          wait_for { `pgrep -f libdatadog-crashtracking-receiver`.lines.size }.to eq 1
+
+          crashtracker.stop
+
+          wait_for { `pgrep -f libdatadog-crashtracking-receiver` }.to be_empty
+        end
+      end
+
+      describe '#update_on_fork' do
+        context 'when _native_stop_crashtracker raises an exception' do
+          it 'logs the exception' do
+            logger = Logger.new($stdout)
+            crashtracker = build_crashtracker(logger: logger)
+
+            expect(described_class).to receive(:_native_start_or_update_on_fork) { raise 'Test failure' }
+            expect(logger).to receive(:error).with(/Failed to update_on_fork crash tracking: Test failure/)
+
+            crashtracker.update_on_fork
+          end
+        end
+
+        it 'update_on_fork the crash tracker' do
+          expect(described_class).to receive(:_native_start_or_update_on_fork).with(
+            hash_including(action: :update_on_fork)
+          )
+
+          crashtracker = build_crashtracker
+
+          crashtracker.update_on_fork
+        end
+
+        it 'updates existing crash tracking process after started' do
+          crashtracker = build_crashtracker
+
+          crashtracker.start
+          crashtracker.update_on_fork
+
+          wait_for { `pgrep -f libdatadog-crashtracking-receiver`.lines.size }.to be 1
+
+          tear_down!
+        end
+
+        context 'when multiple instances' do
+          it 'updates existing crash tracking process after started' do
+            crashtracker = build_crashtracker
+            crashtracker.start
+
+            another_crashtracker = build_crashtracker
+            another_crashtracker.update_on_fork
+
+            wait_for { `pgrep -f libdatadog-crashtracking-receiver`.lines.size }.to be 1
+
+            tear_down!
+          end
```

Comment:
Given we have test coverage that:

1. If you have multiple crashtracker instances, only one process gets started
2. When you update on fork, the existing process gets updated

This "multiple instances test" seems to be 1+2, but it doesn't seem to add coverage that we already didn't get with 1 and 2?